### DISCUSSION TITLE
🌍 Mobile Localization

### DISCUSSION THREAD

**Discussion by fuksman** [anyproto/contributors#93] [Announcements]
Dear community, we are excited to announce the launch of the Mobile Localization project!

We are now accepting translations for Anytype Mobile. You can contribute and have your translations included in upcoming releases.

To participate, please visit the Crowdin platform: [https://crowdin.com/project/anytype-mobile](https://crowdin.com/project/anytype-mobile).

We have already added all languages that were previously requested for the Anytype Desktop l10n project. If your language is not listed on Crowdin, please leave a comment here, and we will quickly add it.

For the translation policy, please refer to our documentation: [https://tech.anytype.io/l10n/policy](https://tech.anytype.io/l10n/policy). If you have any unanswered questions, feel free to ask them here in the comments or on Crowdin.

Please note that while Anytype Apps are not yet optimized for right-to-left languages, we do not currently support their translation. 

Stay tuned! 🙌

**Comment by holton-jiang** [comment]
It's fantastic.

**Comment by krzysztof-tys** [comment]
There's form: one, few, many, other ([example](https://crowdin.com/translate/anytype-mobile/49/en-pl?filter=basic&value=0))

How many is few, many?

What 'other' stands for?

**Reply by fuksman** [reply]
Hey @tys404, it depends on the language. 
Please refer to the [Crowdin documentation](https://support.crowdin.com/online-editor/#:~:text=the%20next%20string.-,Plural%20Forms,-Some%20strings%20may) and [this table](https://www.unicode.org/cldr/charts/44/supplemental/language_plural_rules.html) for information about languages.

**Reply by krzysztof-tys** [reply]
Awesome, thx

**Comment by e4eternal** [comment]
When will you add RTL support? I want to start translating Anytype into Arabic!

**Reply by fuksman** [reply]
It's not clear yet, but we keep this in mind.

**Comment by hlfernandez** [comment]
Hi, I would like to start translating it to [Galician](https://en.wikipedia.org/wiki/Galician_language). Could you add it at Crowdin?

**Reply by fuksman** [reply]
✅ Done

**Comment by eud3** [comment]
Hello, I would like to translate the desktop app to Catalan. Could you add it please? Thank ou

**Reply by fuksman** [reply]
✅ Done

**Comment by bloodwiing** [comment]
Hey there, would like to try to translate Anytype to Lithuanian. Could you please add it as an option in Crowdin?

**Reply by fuksman** [reply]
✅ Done

**Comment by kerimabdy** [comment]
Hi! I would like to translate Anytype to Turkmen. Could you please add it as an option in Crowdin?

**Reply by fuksman** [reply]
✅ Done

**Comment by kapatych-be** [comment]
Please add the Belarusian language, I will take care of the mobile version as soon as I finish with the desktop version.

**Reply by fuksman** [reply]
✅ Done

**Comment by mottcha** [comment]
Hi! Could you please add Japanese? I would like to translate the mobile version after I'm done with the desktop version.

**Reply by mottcha** [reply]
Hi, @fuksman! I'm following up on my request to add Japanese to Crowdin for the mobile version. I've nearly finished the desktop translation and would love to start on mobile. Do you have any plans to add it, or is it on the roadmap?

**Reply by fuksman** [reply]
Hey @mottcha. Sorry for missing this. I've enabled Japanese for translation for mobile. Thanks for your contributions!

**Reply by mottcha** [reply]
Thanks so much! I'll get started on the mobile translation.

Also, when you have a moment, it would be a great help if you could take a look at the hidden strings. A few of them aren't showing up on the desktop translations in the Crowdin editor, like `usecaseCategoryHome` and `emojiCategoryFlags`. For example, `emojiCategoryFlags `and `electronMenuFlags `(related to debugging) are being treated as the same strings, which makes translation difficult. It seems like duplicate words in the original English source file might be causing them to be treated as duplicate strings. This page might be helpful: https://support.crowdin.com/string-management/

**Reply by mottcha** [reply]
Sorry to bother you again, but I've encountered more "hidden strings" while translating Anytype Mobile. I'm having trouble translating important phrases like `TextStyle.Title.Title` and `TextStyle.Underline.Title`, as they are not appearing in the Crowdin editor. It would be a great help if you could look into fixing these.

I had on Localizable.xcstrings at Modules in anytype-swift / develop

The image below is when I tried importing TextStyle.Title.Title using a file.
![test](https://github.com/user-attachments/assets/c76737c2-4964-42e7-a5d8-f5255aa5415f)

**Reply by mottcha** [reply]
Hi @fuksman,
(cc: @ignatovv - Apologies for the direct mention, but I wanted to bring this to your attention as well, as we are eager to see Japanese support on iOS.)

Thank you for adding Japanese to the Android app. Most of the Japanese translation has been completed since May, and I also requested iOS support on the forum, but it has not been added yet. 

Could you please consider enabling the Japanese localization on iOS as well? It would really help maintain motivation to keep the translations up to date.

Thank you for your time and great work.

https://community.anytype.io/t/add-japanese-localization-to-ios-app-translations-ready/27972

**Comment by UnToobed** [comment]
Hi! I would love to contribute to Anytype with a Swedish translation for the mobile app.

**Comment by Zavesky** [comment]
Hi, can you add czech language for mobile app? Thanks

**Reply by ignatovv** [reply]
Hi, we have Czech language in our localization platform, we will release it as soon it will be finished. We use Crowdin as localization platform so this not really in our sphere of control when it will be done tho.
